docs/searching-algorithms/linear-search.mdx

```diff
@@ -0,0 +1,181 @@
+
+ The key is compared to the first element arr[0]=1
+ The key does not match  arr[0]
+
+ Next key is compared to arr[1] = 3
+ Key = 5 does not match the arr[1]
+
+ Next key is compared to arr[2] = 5
+ Key matches arr[2] and index 2 is returned
+
+```
+### Example 2
+
+```
+ Given the below array arr[4]:        
+ | 1 | 3 | 5 | 7 |
+
+ The key element = 9
+
+ The key is compared to the first element arr[0]=1
+ The key does not match  arr[0]
+
+ Next key is compared to arr[1] = 3
+ Key = 5 does not match the arr[1]
+
+ Next key is compared to arr[2] = 5
+ Key = 5 does not match the arr[2]
+ 
+ Next key is compared to arr[3] = 7
+ Key = 5 does not match the arr[3]
+
+ The end of the array is reached. 
+ The key is not present in the array
+ -1 is returned indicating the absence of the key in array 
+
+```
+
+# Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="js"
+  values={[
+    { label: 'JavaScript', value: 'js', },
+    { label: 'Python', value: 'py', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+
+//Linear Search Algorithm
+#include<iostream>
+using namespace std;
+
+int main()
+{
+  int n,i,a;
+  // initialize position variable as -1
+  int pos=-1;    
+  cout<<"Enter number of elements: ";
+  cin>>n;
+  cout<<"Enter the elements "<<endl;
+  int arr[n];
+
+  for(i=0;i<n;i++)
+  {
+    cin>>arr[i];
+  }
+  cout<<"Enter element to search ";
+  cin>>a;
+  
+  //loop from 0th to nth element of array
+  for(i=0;i<n;i++)    
+  {
+    //check whether number is present
+    if(arr[i]==a)       
+    {
+      //if found update pos to i i.e. position of the element
+      pos = i;       
+      cout<<"Found at "<<pos;
+      break;       
+    }
+  }
+   // if pos=-1 then element is not found
+  if(pos == -1)   
+  {
+    cout<<"Not Found";
+  }
+  return 0;
+}
+
+```
+</TabItem>
+
+<TabItem value="py">
+
+```python   
+
+def linearSearch(arr, x):
+    pos = -1
+    for i in range(len(arr)):
+        if arr[i] == x:
+            pos = i
+            print("Found at {}".format(pos))
+            
+    if pos == -1:
+        print("Not Found")
+
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
+
+
+## Sample 1 for the C++
+
+### Input :
+
+```
+Enter number of elements:5
+Enter the elements 1 2 4 3 5
+Enter element to search 3
+```
+### Output :
+
+```
+Found at 3
+
+```
+
+# Time & Space Complexity
+
+Linear search gives linear access to the elements. In the best case, when the element to be searched is present in the first position, then only one iteration will be required. In the worst case when the element to be searched is present in the last position, linear search will have to go through all the elements. Therefore its time complexity is O(n).
+
+In linear search, no extra space is used to store anything. So the space complexity is O(1).
+
+Time Complexity: O(n)
+Time Complexity: O(1)
```

Comment:
It should be `Time & Space` complexity not Time and Time complexity. Also use bulletpoints here